README.md

```diff
@@ -1,8 +1,125 @@
+
+The simplest form for importing a compatible CSV file is:
+
+```rb
+SolidusImporter.import! 'customer-list.csv', type: :customers
+```
+
+### Accepted Format
+
+The accepted format is the [Shopify CSV](https://help.shopify.com/en/manual/products/import-export/using-csv) for which is also relatively easy to find exporters for every major platform (e.g. [shopify_transporter](https://github.com/Shopify/shopify_transporter)).
+
+There are three supported CSV types:
+
+1. [Product](https://help.shopify.com/en/manual/migrating-to-shopify/transporter-app/csv-products)
+2. [Order](https://help.shopify.com/en/manual/migrating-to-shopify/transporter-app/csv-orders)
+3. [Customer](https://help.shopify.com/en/manual/migrating-to-shopify/transporter-app/csv-customers)
+
+### The Processors
+
+The importing is managed by a list of processors for each CSV type, the default processors are:
+
+```rb
+Spree.config do |config|
+  config.importer.product_processors = %w[
+    SolidusImporter::Processors::Product
+    SolidusImporter::Processors::Variant
+    SolidusImporter::Processors::ProductImage
+    SolidusImporter::Processors::VariantImage
+    SolidusImporter::Processors::Log
+  ]
+
+  config.importer.order_processors = %w[
+    SolidusImporter::Processors::Order
+    SolidusImporter::Processors::Log
+  ]
+
+  config.importer.customer_processors = %w[
+    SolidusImporter::Processors::Customer
+    SolidusImporter::Processors::Log
+  ]
+end
+```
+
+Each processor is a callable that will accepts a context Hash. It will perform its function within the `#call(context)` method and will return an equally valid context Hash. The returned context can be augmented with additional data.
+
+Example:
+
+```rb
+CUSTOM_LOGGER = Logger.new(Rails.root.join('log/importer.log'))
+
+CustomLoggerProcessor = ->(context) {
+  context.merge(logger: CUSTOM_LOGGER)
+}
+
+# Replace the original Log processor with CustomLoggerProcessor
+Solidus.configuration.importer.customer_processors.map! do |processor|
+  if processor == 'SolidusImporter::Processors::Log'
+    'CustomLoggerProcessor'
+  else
+    processors
+  end
+end
+```
+
+Each list of processors can be configured to add, remove, or replace any of the default processors.
+
+### The Context Hash
+
+The context hash is required to have some keys:
+
+- `:data` this key will initially contain the raw data coming from the CSV and is generally expected to be present in the returned context (although it can be normalized and repaired)
+- `:logger` is expected to be a logger responding to the standard `Logger` interface
+
+Other than that processors will typically add their own keys and register on the context the records for which they are responsible. For example the Product processor will add a `:product` key that will contain the instance of `Spree::Product` that has been fetched or created.
```

Comment:
"Other than those processors" - I'm using Grammarly, just for minor improvements.